cmd/juju/application/bundle.go

```diff
@@ -1330,11 +1330,14 @@ func buildModelRepresentation(
 				Machine: unit.Machine,
 			})
 		}
-		// TODO: application constraints
 		applications[name] = application
 		annotationTags = append(annotationTags, names.NewApplicationTag(name).String())
 		appNames = append(appNames, name)
-		if len(appStatus.SubordinateTo) == 0 {
+		if len(appStatus.Units) > 0 {
```

Comment:
Reading the code, it looks like doing this will prevent any constraints associated with the app from being processed.  Not sure that's a good thing.

Reply:
That isn't the case here. This is only building the memory representation of the model as it exists for working out what commands to run. If the app is specified in the bundle and has constraints, the worst that will happen is that it tries to set the constraints to what they are already.